circus-train-core/src/main/java/com/hotels/bdp/circustrain/core/event/MetricsListener.java

```diff
@@ -80,8 +80,10 @@ public void tableReplicationFailure(EventTableReplication tableReplication, Stri
 
   @Override
   public void copierEnd(Metrics metrics) {
-    runningMetricsReporter.report();
-    runningMetricsReporter.stop();
+    if (runningMetricsReporter != null) {
```

Comment:
This hides a bug maybe better throw an IllegalStateException when null?

Reply:
+1